harness/nativeFunctionMatcher.js

```diff
@@ -3,17 +3,198 @@
 /*---
 description: Assert _NativeFunction_ Syntax
 info: |
-    This regex makes a best-effort determination that the tested string matches
-    the NativeFunction grammar production without requiring a correct tokeniser.
-
     NativeFunction :
-      function _IdentifierName_ opt ( _FormalParameters_ ) { [ native code ] }
+      function _NativeFunctionAccessor_ opt _IdentifierName_ opt ( _FormalParameters_ ) { [ native code ] }
+    NativeFunctionAccessor :
+      get
+      set
 defines:
-  - NATIVE_FUNCTION_RE
   - assertToStringOrNativeFunction
   - assertNativeFunction
+  - validateNativeFunctionSource
 ---*/
-const NATIVE_FUNCTION_RE = /\bfunction\b[\s\S]*\([\s\S]*\)[\s\S]*\{[\s\S]*\[[\s\S]*\bnative\b[\s\S]+\bcode\b[\s\S]*\][\s\S]*\}/;
+
+const validateNativeFunctionSource = function(source) {
+  const UnicodeIDStart = /\p{ID_Start}/u;
```

Comment:
We shouldn't gate testing of `F.p.toString` on implementation of the `\p` feature in RegExps. That would mean that all `F.p.toString` tests would require the `regexp-unicode-property-escapes` feature flag.

Reply:
I recommend generating a **RegExp** that is equivalent to `\p{ID_Start}` using [regexpu‑core](https://www.npmjs.com/package/regexpu-core).

Reply:
That has the slight downside that it will fall out of date with Unicode (as ECMA262 tracks the latest version for this purpose) and need to be updated frequently, but it's the best strategy I can think of.